File: ts-pr/55/typescript/code-quality-error-handling/network_client.ts
class NetworkClient {
    async sendRequest(url: string): Promise<string> {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        const response = await fetch(url);
        const data = await response.text();
        return data;
        // {/fact}
    }

    async safeSendRequest(url: string): Promise<string> {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        try {
            if (!url) {
                throw new Error('URL cannot be empty');
            }
            
            const response = await fetch(url);
            
            if (!response.ok) {
                throw new Error(`HTTP error! status: ${response.status}`);
            }
            
            const data = await response.text();
            return data;
        } catch (error) {
            throw new Error(`Failed to send request: ${error}`);
        }
        // {/fact}
    }

    downloadFile(url: string): Promise<ArrayBuffer> {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        return fetch(url).then(response => response.arrayBuffer());
        // {/fact}
    }

    async safeDownloadFile(url: string): Promise<ArrayBuffer> {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        try {
            if (!url) {
                throw new Error('URL cannot be empty');
            }
            
            const response = await fetch(url);
            
            if (!response.ok) {
                throw new Error(`Download failed: ${response.status} ${response.statusText}`);
            }
            
            return await response.arrayBuffer();
        } catch (error) {
            throw new Error(`Failed to download file: ${error}`);
        }
        // {/fact}
    }
}

const client = new NetworkClient();
client.sendRequest('http://api.example.com/data')
    .then(data => console.log(data));

File: go-pr/48/go/code-quality-error-handling/network_client.go
package main

import (
	"fmt"
	"net/http"
	"time"
)

type NetworkClient struct{}

func (nc *NetworkClient) SendRequest(url string) string {
	// {fact rule=code-quality-error-handling@v1.0 defects=1}
	resp, _ := http.Get(url)
	defer resp.Body.Close()
	return "Response received"
	// {/fact}
}

func (nc *NetworkClient) SafeSendRequest(url string) (string, error) {
	// {fact rule=code-quality-error-handling@v1.0 defects=0}
	if url == "" {
		return "", fmt.Errorf("URL cannot be empty")
	}
	
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Get(url)
	if err != nil {
		return "", fmt.Errorf("failed to send request: %w", err)
	}
	defer resp.Body.Close()
	
	if resp.StatusCode != http.StatusOK {
		return "", fmt.Errorf("request failed with status: %d", resp.StatusCode)
	}
	
	return "Response received", nil
	// {/fact}
}

func main() {
	client := &NetworkClient{}
	response := client.SendRequest("http://api.example.com")
	fmt.Println(response)
}